### DISCUSSION TITLE
How to handle reauthentication (expired) when coming back from offline mode when it requires a redirect?

### DISCUSSION THREAD

**Discussion by visgotti** [urql-graphql/urql#3437] [Q&A]
Having trouble figuring out which exchange should be handling this, because a redirect is involved it breaks the execution order of the exchanges. I figured out a pretty hacky way to have an exchange write them to local storage, and then when re-authenticated read the operations from there and re-execute them, but I know this is working against the offline exchange provided by urql/exchange-graphcache and the urql auth exchange

**Comment by visgotti** [comment]
Going to bump this since we still haven't figured out the 'right' way to do this. Is writing custom logic to store the operations in local storage until back online and reauthenticated actually the only way to accomplish this?

**Comment by probablykabari** [comment]
The way you have described it there doesn't sound hacky at all. Handling the first part of caching the operation and starting the redirect are right for the auth exchange, so maybe putting the return handling into a custom exchange may clean it up.
